File: shapes/TestShapes.java
package shapes;

public class TestShapes {

    public static void main (String[] args) {

        Shape r = new Rectangle(2, 3, "red", true);
        System.out.println(r);
        System.out.println(r.getArea());
        System.out.println(r.getPerimeter());
        System.out.println(r.toString());
        System.out.println();
        
        Shape c = new Circle(2, "blue", false);
        System.out.println(c);
        System.out.println(c.getArea());
        System.out.println(c.getPerimeter());
        System.out.println(c.getColor());
        System.out.println(c.isFilled());
        System.out.println();
        
        Rectangle s = new Square(2, "red", true);
        System.out.println(s);
        System.out.println(s.getArea());
        System.out.println(s.getPerimeter());
        System.out.println(s.getColor());
        System.out.println(s.isFilled());
        System.out.println();

        Shape t = new EquilateralTriangle(2, "red", true);
        System.out.println(t);
        System.out.println(t.getArea());
        System.out.println(t.getPerimeter());
        System.out.println(t.getColor());
        System.out.println(t.isFilled());
        System.out.println();
    }
}

File: PythonShapes/TestShapes.py
from Shape import Shape
from Circle import Circle
from Rectangle import Rectangle
from EquilateralTriangle import EquilateralTriangle
from Square import Square

def testShape(Shape):
    s1 = Shape("red", True)
    s1.getArea()
    s1.getPerimeter()
    print(s1)

def testCircle(Circle):
    c1 = Circle("yellow", False, 5.0)
    c1.getArea()
    c1.getPerimeter()
    print(c1)

def testRectangle(Rectangle):
    r1 = Rectangle("red", True, 1.0, 1.0)
    r1.getArea()
    r1.getPerimeter()
    print(r1)

def testEquilateralTriangle(EquilateralTriangle):
    et1 = EquilateralTriangle("red", True, 1.0)
    et1.getArea()
    et1.getPerimeter()
    print(et1)

def testSquare(Square):
    s1 = Square("red", True, 1.0)
    s1.getArea()
    s1.getPerimeter()
    print(s1)

if __name__ == "__main__":
    testShape(Shape)
    testCircle(Circle)
    testEquilateralTriangle(EquilateralTriangle)
    testRectangle(Rectangle)
    testSquare(Square)
